hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -919,12 +920,32 @@ private boolean isKeyEmpty(OmKeyInfo keyInfo) {
     // underlying table using an iterator. That automatically creates a
     // snapshot of the data, so we don't need these locks at a higher level
     // when we iterate.
+
+    startKey = normalizeListKeyPath(startKey);
+    keyPrefix = normalizeListKeyPath(keyPrefix);
+
     List<OmKeyInfo> keyList = metadataManager.listKeys(volumeName, bucketName,
         startKey, keyPrefix, maxKeys);
     refreshPipeline(keyList);
     return keyList;
   }
 
+  private String normalizeListKeyPath(String keyPath) {
+
+    String normalizeKeyPath = keyPath;
+    if (enableFileSystemPaths) {
+      // For empty strings do nothing.
+      if (StringUtils.isBlank(keyPath)) {
```

Comment:
Is there any reason to keep these two features (return empty string for empty string + keep closing `/`) in here intead of moving to  `OmUtils.normalizeKey(keyPath)`

Reply:
For key ending with "/" after normalizing it will return the key without trailing "/". As here the key intended to be searched is, I have added it back here. In my view removing "/" at the end when the user explicitly asked to search for keys end with "/", it might return results differently, as I don't want to change current semantics.

Example:
the prefix is "/a/b/c/d/" -> after Normalize "/a/b/c/d"
So if we don't add "/" back, it will return keys with prefix "/a/b/c/d", but user intention here is to get keys in "/a/b/c/d/" so not to change semantics for this case handled in this method instead of common method used by all classes.

And also for null not calling OmUtils.normalizeKey(keyPath), the Paths method will fail with NPE, so this is also considered a special case and handled it in this method.

Reply:
Thanks to explain it.

>  the Paths method will fail with NPE, so this is also 

Not a big deal, but if we move the empty check, to `normalizeKey`, the method will be safe forever, and we don't need to do the empty check all the time when we need to call it.

Reply:
BTW (just a discussion not a code review comment): it can be useful to differentiate between normalization (removing/resolving `..`, `//`) and handling closing `/`. My impression is that we need different rules for them.

This is a good example here: you need the first ("normalization") but not the second. (Maybe two different method?)

Reply:
Thanks for clear explanation, I have updated to move null check to normalizeKey method.

BTW (just a discussion not a code review comment): it can be useful to differentiate between normalization (removing/resolving .., //) and handling closing /. My impression is that we need different rules for them.

Updated OmUtils.normalizeKey to take the boolean flag to preseverTralingSlash if true will preserve it. In this way, this method can be used in listKey case with true, rest all code with false. Have a look in to it, and share ur suggestions on the approach.